### PR TITLE
sample: tf-m psa template redefine purpose

### PR DESCRIPTION
**Pull request by mswarowsky**
Renaming the sample to make clear it is a template that tries to the best state of the art security according to the ARM PSA standard.

Ref: NCSDK-18367

Signed-off-by: Markus Swarowsky <markus.swarowsky@nordicsemi.no>

### CODE REVIEW

File: samples/tfm/tfm_psa_template/README.rst [hunk 1]
```diff
@@ -21,13 +22,14 @@ The sample supports the following development kits:
 Overview
 ********
 
-This sample uses Trusted Firmware-M, nRF Secure Immutable bootloader and MCUboot to satisfy the Platform Security Architecture (PSA) Certified Level 2 requirements.
+This sample uses Trusted Firmware-M, nRF Secure Immutable bootloader and MCUboot to demonstrate on how to implement the best practices given in the Arm Platform Security Architecture (PSA) requirements.
+It includes to provision the device with keys and been able to perform an device firmware update.
```
Comment (torsteingrindvik) [inline samples/tfm/tfm_psa_template/README.rst:26]:
```suggestion
It includes provisioning the device with keys and being able to perform a device firmware update.
```

**Review by Vge0rge — APPROVED** [review]

File: samples/tfm/tfm_psa_template/README.rst [hunk 2]
```diff
@@ -21,13 +22,14 @@ The sample supports the following development kits:
 Overview
 ********
 
-This sample uses Trusted Firmware-M, nRF Secure Immutable bootloader and MCUboot to satisfy the Platform Security Architecture (PSA) Certified Level 2 requirements.
+This sample uses Trusted Firmware-M, nRF Secure Immutable bootloader and MCUboot to demonstrate on how to implement the best practices given in the Arm Platform Security Architecture (PSA) requirements.
```
Comment (endre-nordic) [inline samples/tfm/tfm_psa_template/README.rst:25]:
```suggestion
This sample uses Trusted Firmware-M, nRF Secure Immutable bootloader and MCUboot to demonstrate how to implement the best practices given in the Arm Platform Security Architecture (PSA) requirements.
```

File: samples/tfm/tfm_psa_template/README.rst [hunk 3]
```diff
@@ -1,13 +1,14 @@
-.. _tfm_cert:
+.. _tfm_psa_template:
+
+TF-M: PSA Template
```
Comment (mia-ko) [inline samples/tfm/tfm_psa_template/README.rst:3]:
```suggestion
TF-M: PSA template
```

File: samples/tfm/tfm_psa_template/README.rst [hunk 4]
```diff
@@ -1,13 +1,14 @@
...
 
-TF-M: PSA Certified Level 2
 ###########################
```
Comment (joerchan) [inline samples/tfm/tfm_psa_template/README.rst:5]: Not an RST expert, but I think the ### needs to be the same length as the title.

File: samples/tfm/tfm_psa_template/README.rst [hunk 5]
```diff
@@ -1,13 +1,13 @@
...
-###########################
+TF-M: PSA Template
+##################
 
 .. contents::
    :local:
    :depth: 2
 
-This sample is a reference for how to achieve PSA Certified Level 2 using TF-M on nRF devices.
+This sample gives a template on how to follow best practices according to the Arm PSA on nRF devices.
```
Comment (mia-ko) [inline samples/tfm/tfm_psa_template/README.rst:10]:
```suggestion
This sample provides a template for Arm Platform Security Architecture (PSA) best practices on nRF devices.
```

File: samples/tfm/tfm_psa_template/README.rst [hunk 6]
```diff
@@ -21,13 +21,14 @@ The sample supports the following development kits:
 Overview
 ********
 
-This sample uses Trusted Firmware-M, nRF Secure Immutable bootloader and MCUboot to satisfy the Platform Security Architecture (PSA) Certified Level 2 requirements.
+This sample uses Trusted Firmware-M, nRF Secure Immutable bootloader and MCUboot to demonstrate how to implement the best practices given in the Arm Platform Security Architecture (PSA) requirements.
```
Comment (mia-ko) [inline samples/tfm/tfm_psa_template/README.rst:24]:
```suggestion
This sample uses Trusted Firmware-M, nRF Secure Immutable bootloader and MCUboot to demonstrate how to implement the best practices that comply with the Arm PSA requirements.
```

File: samples/tfm/tfm_psa_template/README.rst [hunk 7]
```diff
@@ -39,12 +40,12 @@ Build and flash the provisioning image sample to provision the device with the P
     west build -b nrf5340dk_nrf5340_cpuapp nrf/samples/tfm/provisioning_image -d build_provisioning_image
     west flash --erase -d build_provisioning_image
 
-Build and flash the TF-M PSA Certified Level 2 sample.
+Build and flash the TF-M PSA Template sample.
```
Comment (mia-ko) [inline samples/tfm/tfm_psa_template/README.rst:43]:
```suggestion
Build and flash the TF-M PSA template sample.
```

**Review by mia-ko — CHANGES_REQUESTED** [review]

**Review by endre-nordic — APPROVED** [review]

**Review by mia-ko — APPROVED** [review]

**Pull request merged**